src/box/sql/malloc.c

```diff
@@ -276,7 +288,7 @@ dbMallocRawFinish(sql * db, u64 n)
 	void *p;
 	assert(db != 0);
 	p = sqlMalloc(n);
-	if (!p)
+	if (p == NULL && n != 0)
```

Comment:
Why not just return NULL in the case of `n == 0`?

Reply:
Thet's the point - it's legitimate to receive NULL pointer as result of size=0 allocation (or reallocation) thus we simply return `p` (which is NULL) at the next line